Toolchain/CMake/Scripts/Projects.cmake

```diff
@@ -424,4 +423,84 @@ endfunction()
 function(nos_get_module name version out_target_name)
 	nos_get_package(${name} ${version} ${out_target_name})
 	set(${out_target_name} ${${out_target_name}} PARENT_SCOPE)
+endfunction()
+
+function(nos_get_module_info_by_path path out_name out_version out_json)
+	execute_process(
+		COMMAND ${NOSMAN_EXECUTABLE} --workspace "${NOSMAN_WORKSPACE_DIR}" info "" "" ${path}
+		RESULT_VARIABLE nosman_result
+		OUTPUT_VARIABLE nosman_output
+	)
+	if(nosman_result EQUAL 0)
+		string(STRIP ${nosman_output} nosman_output)
+		set(err_name "")
+		string(JSON module_name ERROR_VARIABLE err_name GET "${nosman_output}" info id name)
+		string(JSON module_version ERROR_VARIABLE err_version GET "${nosman_output}" info id version)
+		message(STATUS "Module at path ${path} is ${module_name} version ${module_version}")
+
+		set(${out_name} ${module_name} PARENT_SCOPE)
+		set(${out_version} ${module_version} PARENT_SCOPE)
+		set(${out_json} ${nosman_output} PARENT_SCOPE)
+	else()
+		nos_fatal_error("Failed to find module info from path ${path}.")
+	endif()
+endfunction()
+
+function(nos_normalize_plugin_name INPUT OUTPUT_VAR)
+    # Step 1: split by '.'
+    string(REPLACE "." ";" PARTS "${INPUT}")
+
+    # Step 2: first item stays lowercase
+    list(POP_FRONT PARTS FIRST)
+    set(RESULT "${FIRST}")
+
+    # Step 3: uppercase the first character of every subsequent part
+    foreach(PART IN LISTS PARTS)
+        string(SUBSTRING "${PART}" 0 1 FIRST_CHAR)
+        string(SUBSTRING "${PART}" 1 -1 REMAINDER)
+        string(TOUPPER "${FIRST_CHAR}" FIRST_CHAR)
+        set(RESULT "${RESULT}${FIRST_CHAR}${REMAINDER}")
+    endforeach()
+
+    # Output to parent scope
+    set(${OUTPUT_VAR} "${RESULT}" PARENT_SCOPE)
+endfunction()
+
+function(nos_find_all_plugin_dependencies json out_target_names out_target_dirs out_target_include_dirs)
```

Comment:
If this does not take transitive dependencies into account, it should be named something like nos_find_immediate_plugin_dependencies